frontend/server/minio-helper.ts

```diff
@@ -37,6 +38,25 @@ export interface MinioClientOptionsWithOptionalSecrets extends Partial<MinioClie
  * @param config minio client options where `accessKey` and `secretKey` are optional.
  */
 export async function createMinioClient(config: MinioClientOptionsWithOptionalSecrets) {
+  // This logic is AWS S3 specific
+  if (isS3Endpoint(config.endPoint)) {
+    try {
+      const credentials = fromNodeProviderChain();
+      const aws_credentials = await credentials();
+      if (aws_credentials) {
+        const {
+          accessKeyId: accessKey,
+          secretAccessKey: secretKey,
+          sessionToken: sessionToken,
+        } = aws_credentials;
+        return new MinioClient({ ...config, accessKey, secretKey, sessionToken });
+      }
+    } catch (err) {
+      console.error('Unable to get credentials from AWS credential provider chain: ', err);
+    }
+  }
+
+  // This logic is S3 generic
   if (!config.accessKey || !config.secretKey) {
     try {
       if (await awsInstanceProfileCredentials.ok()) {
```

Comment:
This code is outdated and should be removed since there is an updated implementation for AWS S3. New implementation supports multiple ways to provide credentials including better security practices.

Better to discuss it in a separate PR